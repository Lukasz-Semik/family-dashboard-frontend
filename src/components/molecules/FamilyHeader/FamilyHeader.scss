.family-header {
  display: flex;
  justify-content: center;
  padding-top: 1rem;
}

.family-header-text {
  margin-right: 2rem;
  font-size: 1.5rem;
  color: $dark-green;

  .label {
    margin-right: 5px;
  }

  .value {
    font-weight: 700;
  }
}
