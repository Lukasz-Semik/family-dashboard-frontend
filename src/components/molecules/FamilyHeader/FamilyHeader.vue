<template>
  <div :class="[$style['family-header']]">
    <p :class="[$style['family-header-text']]">
      <span :class="[$style['label']]">{{ $t('dashboard.family') }}:</span>
      <span :class="[$style['value']]">{{ currentFamily.name }}</span>
    </p>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';

import { currentFamily } from '@/store/currentFamily/getters';

export default {
  computed: {
    ...mapGetters({ currentFamily }),
  },
};
</script>

<style lang="scss" module src="./FamilyHeader.scss" />
