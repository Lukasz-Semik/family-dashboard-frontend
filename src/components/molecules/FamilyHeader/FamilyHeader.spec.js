import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';

import FamilyHeader from './FamilyHeader.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('<FamilyHeader', () => {
  let actions;
  let store;
  let getters;

  beforeEach(() => {
    getters = {
      currentFamily: () => ({ users: [{ firstName: 'John' }, { firstName: 'Jane' }] }),
    };
    store = new Vuex.Store({
      state: {},
      actions,
      getters,
    });
  });

  it('should render proper header', () => {
    const wrapper = shallowMount(FamilyHeader, {
      mocks: {
        $t: key => key,
      },
      store,
      localVue,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
