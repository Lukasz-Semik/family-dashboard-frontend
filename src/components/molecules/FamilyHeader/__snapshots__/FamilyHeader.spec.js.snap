// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FamilyHeader should render proper header 1`] = `
<div class="family-header">
  <p class="family-header-text"><span class="label">dashboard.family.name:</span> <span class="value"></span></p>
  <p class="family-header-text"><span class="label">dashboard.family.size:</span> <span class="value">2</span></p>
</div>
`;
